doc/new-guidelines.md

```diff
@@ -80,6 +80,63 @@ This mistake mostly happens with functional components
 
 it's important to name functional components with `f-` prefix
 
+### Component props and api scheme to match Figma as closely as possible
+
+Ideally, the prop names for components (particularly in quo2 Design System) should match the Figma properties as best 
+as possible. This makes it easier for the developer using that component to then configure it correctly for the screen it is
+being used on and avoids unnecessary overwrites and adjustments being made.
+
+Some examples include
+
+#### Avoid unnecessarily grouping categories to reduce the number of props
+For example in Figma if there is component and it has variants the following variants
+
+|theme: "light" blur: "False"|theme: "dark" blur: "False"|theme: "light" blur: "True"|theme: "dark"  blur: "True"|
+|----------------------------|---------------------------|---------------------------|---------------------------|
+| type :neutral name "ABC"   | type :neutral name "ABC"  |                           |                           |
+| type :active name "ABC"    | type :active name "ABC"   |                           |                           |
+| type :danger name "ABC"    | type :danger name "ABC"   | type :danger name "ABC"   | type :danger name "ABC"   |
+
+```clojure
+;; bad
+"theme - :light or :dark
+type - can be :neutral :active :danger :danger-blur"
+(defn my-component [{:keys [theme type]} name])
+
+;; good
+"theme - :light or :dark
+ type - can be :neutral :active :danger
+ blur? - boolean
+"
+(defn my-component [{:keys [theme blur? type]} name])
+```
+
+Please note this is only for the external api of the component and there should be no restriction of how the component manages
+its internal api as that will not affect the developer using the component with the issues described above.
+
+In some cases this is not always possible or does not make sense. However the thought process should be how easy will it be for another developer 
+to use this component with the correct configuration given the screen designs for Figma.
+
+#### Avoid unnecessarily renaming props 
+In general it can be helpful to avoid renaming props from their counterpart in Figma.
+A very common example of this is with Component sizes.
+
+For example if Figma has sizes 20px, 32px and 40px
+
+```clojure
+;; bad
+":size - :small, :medium or :large"
```

Comment:
I agree with the section recommendation `Avoid unnecessarily renaming props`, but I don't agree with the example.

I personally don't like the fact that if a designer decides to change 32 to 30 (and it will happen, let's just wait...) in the future, we'll have to manually update tons of places. It's a separate discussion, but to me most spacing props exposed by components should have some semantic meaning and the Design System should be changed too (of course, it's the source of truth).

Reply:
hmm, yeah there are many other examples of renaming I can include so I will provide an alternative. 
The size issue is pretty annoying to me though, but perhaps we can figure a better solution out.

Reply:
I understand your point with this example. I guess it's okay to keep it, we should probably discuss this a little bit outside this PR :)

Reply:
I adjusted to something similar but suggests the same idea 👍